File: misc/versus/haskell/blastoff.hs
--  Haskell source code

--  blastoff.hs ~~
--                                                      ~~ (c) SRW, 04 Sep 2011

import System.Environment

main :: IO()
main = do
    argv <- getArgs
    putStrLn (countdown 10)
        where
            countdown :: Int -> String
            countdown 0 = "Blastoff!"
            countdown n = if n > 0
                            then (show n ++ " ... " ++ countdown (n - 1))
                            else (countdown 0)

--  vim:set syntax=haskell:

File: misc/versus/java/Blastoff.java
//- Java source code

//- Blastoff.java ~~
//                                                      ~~ (c) SRW, 05 Sep 2011

public class Blastoff {
  private static String countdown(Integer t) {
    return (t == 0) ? "Blastoff!" : t.toString() + " ... " + countdown(t - 1);
  }
  public static void main(String[] argv) {
    System.out.println(countdown(10));
  }
}

//- vim:set syntax=java:
